Samples/ExtendedControlAndMetadata/EyeGazeAndBackgroundSegmentation/README.md

```diff
@@ -0,0 +1,245 @@
+        }
+        // else there is no frame metadata present
+
+        return result;
+    }
+```
+
+## Eye Gaze Correction
+the eye gaze correction DDI (*```KSPROPERTY_CAMERACONTROL_EXTENDED_EYEGAZECORRECTION```*) has been introduced in Windows strating with build 20348. If supported by the driver, it allows to slighly change the predominant face's eye position so that the main subject appears as if looking directly into the camera.
+Programmatically it acts simply as a ON/OFF toggle by setting these flag values accordingly:
+```
+KSCAMERA_EXTENDEDPROP_EYEGAZECORRECTION_OFF = 0x0000000000000000
+KSCAMERA_EXTENDEDPROP_EYEGAZECORRECTION_ON = 0x0000000000000001
+```
+
+## Background Segmentation control
+the Background segmentation DDI (*```KSPROPERTY_CAMERACONTROL_EXTENDED_BACKGROUNDSEGMENTATION```*) has been introduced in Windows strating with build 20348. In this initial release, if supported by the driver, it allows to blur the background of a scene and preserve only parts of the image where the main subject is present. This is a popular scenario in video teleconferencing applications.
+Programmatically it acts simply as a ON/OFF toggle for background blur by setting these flag values accordingly:
+```
+KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_OFF = 0x0000000000000000
+KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_BLUR = 0x0000000000000001
+```
+
+Additionaly in Windows build above 21364, a new mode has been added that request the driver to attach the segmentation mask as metadata to each frame streamed out. This mode is signaled using this flag value:
+```
+KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_MASK = 0x0000000000000002
+```
+When a GET command is sent for the ```KSPROPERTY_CAMERACONTROL_EXTENDED_BACKGROUNDSEGMENTATION```, only whenever the camera device supports the ```KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_MASK``` mode, it will advise with which stream configurations it supports the generation of the mask metadata. This is usefull for an application wanting to understand if there are any limitations before attempting to rely on a mask to operate scenarios such as alpha blending of the background portion of an image. A camera device that supports this control may not always do so accross all its stream configurations. For example, a camera could be able to segment background at 30fps on a 1080p stream, but not at 60fps eventhough it may be able to stream at such resolution and framerate combination. The structure returned to advise which stream configuration support the generation of the mask frame metadata is a set of ```KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_CONFIGCAPS```. The app can then correlate the available stream configurations of a camera with this set to understand what to expect if it sets this control with the ```KSCAMERA_EXTENDEDPROP_BACKGROUNDSEGMENTATION_MASK``` flag.
+
+## Background Segmentation mask metadata
+The background segmentation mask contains the mask metadata frame as a contiguous buffer,
```

Comment:
>, [](http://example.com/codeflow?start=88&length=1)

nit: this is one of the few places you manually linebreak. Does that show in the final result? Could that cause weird wrap inconsistencies in some widths?

Reply:
don't know why I used a line break, will remove. does not cause visible break in the parsed md, but is not consistent.